powershell/upload/UploadFile.ps1

```diff
@@ -0,0 +1,533 @@
+}
+
+class CredentialsManager {
+    hidden [string] $_credentialsPath
+
+    CredentialsManager([string] $storagePath) {
+        $this._credentialsPath = $storagePath
+    }
+
+    [void] CreateNew() {
+        $storagePath = Split-Path $this._credentialsPath
+        
+        Write-Host "----"
+        Write-Host "Saving your credentials."
+        Write-Host "| Path: $($this._credentialsPath)"
+
+        if (-not (Test-Path -Path $storagePath -PathType Container)) {
+            Write-Host "----"
+            Write-Host "Storage credential filePath doesn't exist. Creating it."
+            Write-Host "| Path: $($storagePath)"
+            
+            New-Item -ItemType Directory -Force -Path $storagePath
+        }
+
+        Write-Host "Enter your credentials."
+        $clientId = Read-Host -Prompt '| Client ID'
+        $clientSecret = Read-Host -Prompt '| Client secret' -AsSecureString
+
+        [PSCredential]::new($clientId, $clientSecret) | Export-CliXml -Path $this._credentialsPath
+
+        Write-Host "----"
+        Write-Host "Credentials saved."
+    }
+
+    [Credentials] Retrieve() {
+        Write-Host "----"
+        Write-Host "Retrieving your credentials."
+        Write-Host "| Path: $($this._credentialsPath)"
+
+        if (-not (Test-Path -Path $this._credentialsPath -PathType Leaf)) {
+            Write-Host "----"
+            Write-Host "Credentials not found."
+            Write-Host "| Path: $($this._credentialsPath)"
+            
+            return $null
+        }
+
+        $credentialsStorage = Import-CliXml -Path $this._credentialsPath
+
+        $credentials = [Credentials]::new()
+        $credentials.ClientId = $credentialsStorage.GetNetworkCredential().UserName
+        $credentials.ClientSecret = $credentialsStorage.GetNetworkCredential().Password
+
+        Write-Host "Credentials retrieved."
+
+        return $credentials
+    }
+}
+
+class FileApiService {
+    hidden [FileApiClient] $_fileApiClient
+    hidden [string] $_tenantId
+    hidden [long] $_uploadSizeLimit
+    hidden [string] $_boundary
+    hidden [string] $_businessTypeId
+
+    FileApiService(
+        [FileApiClient] $fileApiClient,
+        [string] $tenantId,
+        [string] $businessTypeId
+    ) {
+        $this._fileApiClient = $fileApiClient
+        $this._tenantId = $tenantId
+        $this._boundary = "file_info"
+        $this._businessTypeId = $businessTypeId
+
+        # API supports files up to 100 megabytes
+        $this._uploadSizeLimit = 100 * 1024 * 1024
+    }
+
+    [string] CreateFileToUpload([string] $contentFilePath) {
+        $headerFilePath = ""
+        $footerFilePath = ""
+        try {
+            $folderPath = $(Split-Path -Path $contentFilePath)
+            $contentFilename = $(Split-Path -Path $contentFilePath -Leaf)
+            $createdFilePath = "$($folderPath)\$([Helper]::ConverToUniqueFileName("multipart.bin"))"
+
+            $headerFilename = [Helper]::ConverToUniqueFileName("header.txt")
```

Comment:
Maybe $headerFile**N**ame instead of $headerFilename?

Reply:
I will change all fileName to filename.